meilisearch-core/src/query_builder.rs

```diff
@@ -250,15 +257,19 @@ impl<'c, 'f, 'd, 'i> QueryBuilder<'c, 'f, 'd, 'i> {
             result.retain(|doc| {
                 let id = doc.id;
                 let key = (distinct)(id);
-                match key {
+                let distinct_accepted = match key {
                     Some(key) => distinct_map.register(key),
                     None => distinct_map.register_without_key(),
+                };
+                if distinct_accepted {
```

Comment:
is it not the opposite?
```suggestion
                if !distinct_accepted {
```

Reply:
Fixed.